src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -531,32 +624,70 @@ public String getQuestionAdditionalInfoHtml(int questionNumber, String additiona
     }
 
     @Override
-    public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> responses,
+    public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> unsortedResponses,
             FeedbackQuestionAttributes question,
             String studentEmail,
             FeedbackSessionResultsBundle bundle,
             String view) {
 
-        if ("student".equals(view) || responses.isEmpty()) {
+        if ("student".equals(view) || unsortedResponses.isEmpty()) {
             return "";
         }
 
-        Map<String, Integer> answerFrequency = new LinkedHashMap<>();
-        int numChoicesSelected = getNumberOfResponses(responses, answerFrequency);
+        MSQStatistics msqStats = new MSQStatistics(this);
+        // Sort responses based on recipient team and recipient name.
+        List<FeedbackResponseAttributes> responses = msqStats.getResponseAttributesSorted(unsortedResponses, bundle);
```

Comment:
Sorting is only needed for `per recipient statistics table`?
So put is in that block? Same for MCQ question?

Reply:
Yes, you are right. Will put this in the per recipient block.